jakarta-ee-messaging/src/main/kotlin/io/qalipsis/examples/jakarta/JakartaProduceAndConsumeFromQueue.kt

```diff
@@ -0,0 +1,133 @@
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import io.kotest.assertions.asClue
+import io.kotest.assertions.assertSoftly
+import io.kotest.matchers.ints.shouldBeExactly
+import io.qalipsis.api.annotations.Scenario
+import io.qalipsis.api.executionprofile.regular
+import io.qalipsis.api.scenario.scenario
+import io.qalipsis.api.steps.filterNotNull
+import io.qalipsis.api.steps.innerJoin
+import io.qalipsis.api.steps.map
+import io.qalipsis.api.steps.verify
+import io.qalipsis.examples.utils.BatteryState
+import io.qalipsis.examples.utils.ScenarioConfiguration
+import io.qalipsis.examples.utils.ServerConfiguration
+import io.qalipsis.plugins.jackson.csv.csvToObject
+import io.qalipsis.plugins.jackson.jackson
+import io.qalipsis.plugins.jakarta.consumer.consume
+import io.qalipsis.plugins.jakarta.deserializer.JakartaJsonDeserializer
+import io.qalipsis.plugins.jakarta.destination.Queue
+import io.qalipsis.plugins.jakarta.jakarta
+import io.qalipsis.plugins.jakarta.producer.JakartaMessageType
+import io.qalipsis.plugins.jakarta.producer.JakartaProducerRecord
+import io.qalipsis.plugins.jakarta.producer.produce
+import jakarta.jms.Session
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory
+
+class JakartaProduceAndConsumeFromQueue {
+
+    private val objectMapper = ObjectMapper().also {
+        it.registerModule(JavaTimeModule())
+    }
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @Scenario("jakarta-produce-and-consume-from-queue")
+    fun scenarioProduceAndConsumeFromQueue() {
+
+        //we define the scenario, set the name, number of minions and rampUp
+        scenario {
+            minionsCount = ScenarioConfiguration.NUMBER_MINION
+            profile {
+                regular(periodMs = 1000, minionsCountProLaunch = minionsCount)
+            }
+        }
+            .start()
+            .jackson() //we start the jackson step to fetch data from the csv file. we will use the csvToObject method to map csv entries to list of utils.BatteryState object
+            .csvToObject(BatteryState::class) {
+
+                classpath("battery-levels.csv")
+                // we define the header of the csv file
+                header {
+                    column("deviceId")
+                    column("timestamp")
+                    column("batteryLevel").integer()
+                }
+                unicast()
+            }
+            .map { it.value } // we transform the output of the CSV reader entries to utils.BatteryState
+            .jakarta()
+            .produce {
+
+                connect {
+                    ActiveMQConnectionFactory(
+                        ServerConfiguration.SERVER_URL,
+                        ServerConfiguration.CONTAINER_USERNAME,
+                        ServerConfiguration.CONTAINER_PASSWORD
+                    ).createQueueConnection()
+                }
+
+                session {
```

Comment:
normally, the connection should be provided as argument `it`, so that you do not create a new one.  If not, we need to change the DSL to allow it. It does not make any sense to create the connection twice.